h/api/nipsa/search.py

```diff
@@ -0,0 +1,83 @@
+import copy
+
+
+def nipsa_filter(query, userid=None):
```

Comment:
It's _filters_ that are encapsulated here, `api/search.py` knows about Es queries of course but it doesn't know anything about filters yet, including how to insert them into query dicts, all that knowledge is in this one function in `api/nipsa/search.py`.

Filtering was something I was only introducing into the code for nipsa so for the sake of not adding new "generic" functions into other parts of the code that are only used for the nipsa function, and to minimize nipsa's footprint outside of `h/api/nipsa/` (that footprint being just a couple of one-liners was one of my favourite things about this approach to nipsa), I encapsulated it here.

I imagine once we have nipsa and groups, two features that are probably both gonna use filters, then we'll need some generic inserting filters in queries code in `api/search.py`, though. I would wait until we have two users but I can do it now if you really want